### ISSUE TITLE
Mastodon link bug

### ISSUE THREAD

**Issue by marcelsalathe** [quarto-dev/quarto-cli#5035]
### Bug description

Big quarto fan here - congrats on building such a fabulous system.

I'm running into a small Mastodon issue.

The following

```
  navbar:
    right:
      - icon: mastodon
        href: https://mastodon.social/@marcelsalathe
        aria-label: Mastodon
```

Renders as 
```
<a class="nav-link" href="https://mastodon.social/(marcelsalathe?)" rel="" target=""><i class="bi bi-mastodon" role="img" aria-label="Mastodon">
```

Everything nice, except the broken URL.

I could add `ref: https://mastodon.social/%40marcelsalathe` (this is what I currently do at [https://www.digitalepibook.com/](https://www.digitalepibook.com/) but I'd prefer the original URL to render correctly in the HTML source code.

My relevant quarto check:
```
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.3.272
      Path: /Applications/quarto/bin

[✓] Checking basic markdown render....OK

```
I'm on MacOS 12.5.1

### Checklist

- [x] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [x] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [x] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [x] Please document the operating system you're running. If on Linux, please provide the specific distribution.
- [x] Please provide the output of `quarto check` so we know which version of quarto and its dependencies you're running.

**Comment by cderv** [comment]
> [✓] Checking Quarto installation......OK
      Version: 1.3.272

Can you try with latest version of Quarto 1.3 ? 

This renders correctly for me 
````html
<a class="nav-link" href="https://mastodon.social/@marcelsalathe" rel="" target=""><i class="bi bi-mastodon" role="img" aria-label="Mastodon"></i><span class="menu-text"></span></a>
````

Other idea: it is possible that we resolve this as a citation ? 🤔 

Do you have reference activated for citation in your website ? Can you share an example of your project ? 

I tried with a simple demo website FWIW

````
quarto create-project mywebsite --type website
```` 
and added in `_quarto.yml`
````yaml
    right:
      - icon: mastodon
        href: https://mastodon.social/@marcelsalathe
        aria-label: Mastodon
````

under `navbar`.

**Comment by dragonstyle** [comment]
Related #3493 - I think the issue is very much that citeproc 'sees' this as a citation. I don't know a way to easily dodge this one :(

**Comment by allenmanning** [comment]
@marcelsalathe Can the [work-arounds documented in this issue](https://github.com/quarto-dev/quarto-cli/issues/3493) help unblock you while we work on our fix?

**Comment by marcelsalathe** [comment]
@allenmanning absolutely. This is not a blocking issue for me. Thanks!

**Comment by cderv** [comment]
Duplicate of #3493

**Comment by cderv** [comment]
@marcelsalathe I'll close this as a duplicate, and we can follow in #3493
